src/PackageJsonSynchronizer.php

```diff
@@ -24,22 +24,30 @@ class PackageJsonSynchronizer
 {
     private $rootDir;
     private $vendorDir;
+    private $scriptExecutor;
     private $versionParser;
 
-    public function __construct(string $rootDir, string $vendorDir = 'vendor')
+    public function __construct(string $rootDir, string $vendorDir, ScriptExecutor $scriptExecutor)
     {
         $this->rootDir = $rootDir;
         $this->vendorDir = $vendorDir;
+        $this->scriptExecutor = $scriptExecutor;
         $this->versionParser = new VersionParser();
     }
 
     public function shouldSynchronize(): bool
     {
-        return $this->rootDir && file_exists($this->rootDir.'/package.json');
+        return $this->rootDir && (file_exists($this->rootDir.'/package.json') || file_exists($this->rootDir.'/importmap.php'));
     }
 
     public function synchronize(array $phpPackages): bool
     {
+        if (file_exists($this->rootDir.'/importmap.php')) {
+            $this->synchronizeForAssetMapper($phpPackages);
+
+            return false;
```

Comment:
should we configure also package.json in case both exist?

Reply:
It's a good question - e.g. "if `package.json` exists, why not update it?". But one use-case will be that someone is using AssetMapper but has a small package.json file so that they can process Tailwind files or React files. In that case, we wouldn't want to update their `package.json`. So my thinking currently is: there's a boolean choice between Encore vs AssetMapper / `importmap.php` vs `package.json`.